NFT/src/interface.sw

```diff
@@ -0,0 +1,229 @@
+library interface;
+
+dep data_structures;
+
+use data_structures::TokenMetaData;
+use std::{identity::Identity, option::Option};
+
+pub struct AdminEvent {
+    /// The user which is now the admin of this contract.
+    /// If there is no longer an admin then the `Option` will be `None`.
+    admin: Option<Identity>,
+}
+
+pub struct ApprovalEvent {
+    /// The user that has gotten approval to transfer the specified token.
+    /// If an approval was revoked, the `Option` will be `None`.
+    approved: Option<Identity>,
+
+    /// The user that has given or revoked approval to transfer his/her tokens.
+    owner: Identity,
+
+    /// The unique identifier of the token which the approved may transfer.
+    token_id: u64,
+}
+
+pub struct BurnEvent {
+    /// The user that has burned their token.
+    owner: Identity,
+
+    /// The unique identifier of the token which has been burned.
+    token_id: u64,
+}
+
+pub struct MintEvent {
+    /// The owner of the newly minted tokens.
+    owner: Identity,
+
+    /// The starting range of token ids that have been minted in this transaction.
+    token_id_start: u64,
+
+    /// The total number of tokens minted in this transaction.
+    total_tokens: u64,
+}
```

Comment:
Is there a reason you moved the events into the contract interface?  I don't think this is a bad thing necessarily, but I think we should be consistent about how we structure our applications.  The reason being is it will be easier for new developers to find code they are looking for if our applications are structured similarly.

Reply:
Since a lot of other applications will be importing the NFT interface they may need access to events hence why they are moved here. I will be adding an NFT interface to the sway-libs repository and then refactoring this application to use it once it is implemented which should clean this up.